web_interface/blueprints/api_v3.py

```
@@ -1559,6 +1559,20 @@ def execute_system_action():
+                # plugin-repos/) even if the user uninstalled them. Re-remove
+                # any the user previously uninstalled so the update doesn't
+                # resurrect them.
+                if api_v3.plugin_store_manager:
+                    try:
+                        purged = api_v3.plugin_store_manager.purge_uninstalled_plugins()
+                        if purged:
+                            logger.info(
+                                "Re-removed %d uninstalled plugin(s) restored by update: %s",
+                                len(purged), ", ".join(purged),
+                            )
+                    except Exception as purge_err:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)